File: 0078/solution.py
class Solution:
    def subsets(self, nums: list[int]) -> list[list[int]]:
        res = []
        curr = []

        def backtrack(i):
            if i == len(nums):
                res.append(curr.copy())
                return
            curr.append(nums[i])
            backtrack(i + 1)
            curr.pop()
            backtrack(i + 1)

        backtrack(0)
        return res

File: 0078/solution.ts
function subsets(nums: number[]): number[][] {
  const res: number[][] = [];
  const curr: number[] = [];
  const backtrack = (index: number) => {
    res.push(curr.slice());
    if (curr.length === nums.length) return;
    for (let i = index; i < nums.length; i++) {
      curr.push(nums[i]);
      backtrack(i + 1);
      curr.pop();
    }
  };
  backtrack(0);
  return res;
}
